### PR TITLE
live tests with some oauth action

### PR DESCRIPTION
**Pull request by andymckay**
Also fixes 403 handling, which was previously throwing a 500 error, not a 403.

Secrets are not available on pull requests, so the live tests will fail until we merge this.

### CODE REVIEW

File: solitude/settings/base.py [hunk 1]
```diff
@@ -355,9 +355,18 @@
     '--logging-filter=xsd',
     '--with-nicedots',
     '--with-blockage',
-    '--http-whitelist=""',
 ]
 
+if not os.environ.get('LIVE_TESTS'):
```
Comment (kumar303) [inline solitude/settings/base.py]: I don't think you can simply bool the value. Consider this:
````
>>> bool("0")
True
````

Comment (kumar303) [inline solitude/settings/base.py]: but maybe that's ok. We wouldn't want the code to get too ugly.

File: solitude/tests/live.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+for key in envs.keys():
+    value = ''
+    if local:
+        log.info('Found {0} from local.py.'.format(key))
+        value = getattr(local, key)
+
+    if not value:
+        log.info('Found {0} from environment.'.format(key))
+        value = os.environ.get(key)
```
Comment (kumar303) [inline solitude/tests/live.py]: Why is this necessary? The [base settings](https://github.com/andymckay/solitude/blob/378/solitude/settings/base.py) file already gets values from the environment like ``BRAINTREE_MERCHANT_ID = os.getenv('BRAINTREE_MERCHANT_ID', '')``

File: solitude/settings/test.py [hunk 3]
```diff
@@ -55,6 +55,10 @@
 
 BOKU_SECRET_KEY = 'for-testing-only'
 
+BRAINTREE_MERCHANT_ID = 'testing-merchant-id'
+BRAINTREE_PUBLIC_KEY = 'testing-public-key'
+BRAINTREE_PRIVATE_KEY = 'testing-private-key'
```
Comment (kumar303) [inline solitude/settings/test.py]: Why do these need to be set?

File: solitude/tests/live.py (hunk 2 above)
Comment (kumar303) [inline solitude/tests/live.py]: In other words, if the value is not defined in a *local* settings file then you could just use the base value

Comment (kumar303) [inline solitude/tests/live.py]: I suppose the ``settings_test.py`` values are the cause of why you have to look for environment variables again. What if ``settings_test.py`` didn't override anything? We have nose blockage.

File: solitude/settings/base.py (hunk 1 above)
Comment (andymckay) [inline solitude/settings/base.py]: We could test specific values, but testing for the presence of the env is enough and pretty common.

File: solitude/settings/test.py (hunk 3 above)
Comment (andymckay) [inline solitude/settings/test.py]: Can't spot a reason now, will remove.

File: solitude/tests/live.py (hunk 2 above)
Comment (andymckay) [inline solitude/tests/live.py]: I can't find a reason now either. Will remove.

File: solitude/tests/live.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+from nose.plugins.attrib import attr
+
+from curling import lib
+
+from solitude.base import getLogger
+
+log = getLogger('s.tests')
+
+configs = {
```
Comment (kumar303) [inline solitude/tests/live.py]: could call this ``overrides`` too

**Comment by kumar303** [comment]
looks good, r+
